tests/test_server_startup.py

```
@@ -0,0 +1,123 @@
+             patch("synapse.server.send_initial_instructions") as mock_instructions:
+
+            # Setup mocks
+            mock_load.return_value = {
+                "command": "echo",
+                "args": ["hello"],
+                "idle_regex": "> ",
+                "env": {"TEST_VAR": "1"},
+                "submit_sequence": "\\n"
+            }
+            
+            mock_reg.return_value = mock_reg_instance
```

Ruff (0.14.10)

Undefined name `mock_reg_instance`

(F821)